subtype/src/main/java/com/fasterxml/jackson/module/subtype/SubtypeModule.java

```diff
@@ -0,0 +1,120 @@
+package com.fasterxml.jackson.module.subtype;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.introspect.Annotated;
+import com.fasterxml.jackson.databind.jsontype.NamedType;
+import com.fasterxml.jackson.module.subtype.PackageVersion;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Subtype module.
+ * <p>
+ * The module caches the subclass, so it's non-real-time.
+ * It's for registering subtypes without annotating the parent class.
+ * See <a href="https://github.com/FasterXML/jackson-databind/issues/2104">this issues</a> in jackson-databind.
+ * <p>
+ * When not found in the cache, it loads and caches subclasses using SPI.
+ * Therefore, we can {@link #unregisterType} a class and then module will reload this class's subclasses.
+ */
+public class SubtypeModule extends Module {
```

Comment:
```suggestion
 * Therefore, we can {@link #unregisterType} a class and then module will reload this class's subclasses.
 *
 * @since 2.16
 */
public class SubtypeModule extends Module {
```